File: Inboxes/Python/Cli/tests/test_inboxes.py
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from inboxes import send_message, list_messages, read_message, delete_message, messages, next_id


def setup_function():
    messages.clear()
    import inboxes
    inboxes.next_id = 1


def test_send_message():
    msg = send_message("alice", "Hello", "World")
    assert msg["id"] == 1
    assert msg["from"] == "alice"
    assert msg["subject"] == "Hello"
    assert msg["body"] == "World"
    assert msg["read"] is False
    assert msg["created_at"] is not None


def test_list_messages():
    send_message("alice", "Subject1", "Body1")
    send_message("bob", "Subject2", "Body2")
    msgs = list_messages()
    assert len(msgs) == 2


def test_read_message_marks_as_read():
    send_message("alice", "Test", "Body")
    msg = read_message(1)
    assert msg is not None
    assert msg["read"] is True
    msg2 = read_message(1)
    assert msg2["read"] is True


def test_delete_message():
    send_message("alice", "Del", "Me")
    assert len(list_messages()) == 1
    assert delete_message(1) is True
    assert len(list_messages()) == 0


def test_list_after_delete():
    send_message("alice", "Keep", "Me")
    send_message("bob", "Delete", "This")
    delete_message(2)
    msgs = list_messages()
    assert len(msgs) == 1
    assert msgs[0]["id"] == 1


def test_read_nonexistent():
    assert read_message(999) is None


def test_delete_nonexistent():
    assert delete_message(999) is False

File: Inboxes/Ruby/Cli/tests/test_inboxes.rb
require 'minitest/autorun'
require_relative '../inboxes'

class TestInboxes < Minitest::Test
  def setup
    $messages.clear
    $next_id = 1
  end

  def test_send_message
    msg = send_message('Alice', 'Hello', 'World')
    assert_equal 1, msg[:id]
    assert_equal 'Alice', msg[:from]
    assert_equal 'Hello', msg[:subject]
    assert_equal 'World', msg[:body]
    assert_equal false, msg[:read]
    refute_nil msg[:created_at]
  end

  def test_increment_id
    send_message('A', 'S1', 'B1')
    msg2 = send_message('B', 'S2', 'B2')
    assert_equal 2, msg2[:id]
  end

  def test_list_messages
    send_message('A', 'S', 'B')
    assert_equal 1, list_messages.length
  end

  def test_read_message_marks_read
    msg = send_message('A', 'S', 'B')
    read = read_message(msg[:id])
    assert_equal msg[:id], read[:id]
    assert_equal true, read[:read]
  end

  def test_read_message_not_found
    assert_nil read_message(999)
  end

  def test_delete_message
    msg = send_message('A', 'S', 'B')
    assert_equal true, delete_message(msg[:id])
    assert_equal 0, list_messages.length
  end

  def test_delete_message_not_found
    assert_equal false, delete_message(999)
  end
end